eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/tasklets/reportCheck/ReportCheckTasklet.java

```diff
@@ -0,0 +1,235 @@
+                boolean removed = variantBuffer.remove(reportedVariant);
+                if (!removed) {
+                    newUnmatchedAccessions.add(reportedVariant);
+                }
+            }
+        }
+        accessionBuffer.addAll(newUnmatchedAccessions);
+        return reportVariantsRead == null;
+    }
+
+    private void removeRemainingMatchingVariants(Set<Variant> variantBuffer,
+                                                 Set<Variant> accessionBuffer) throws Exception {
+        removeMatchingVariants(variantBuffer, accessionBuffer);
+        removeRemainingMatchingVariantsFromOriginalVcfFile(variantBuffer, accessionBuffer);
+    }
+
+    private void removeRemainingMatchingVariantsFromOriginalVcfFile(Set<Variant> variantBuffer,
+                                                                    Set<Variant> accessionBuffer) throws Exception {
+        List<Variant> variantsRead;
+        while ((variantsRead = readVcfIgnoringNonVariants()) != null) {
+            for (Variant variantWithoutAccession : variantsRead) {
+                boolean removed = accessionBuffer.remove(variantWithoutAccession);
+                if (!removed) {
+                    variantBuffer.add(variantWithoutAccession);
+                }
+            }
+        }
+    }
+
+    private void logStatus(StepContribution stepContribution, Set<Variant> variantsWithoutAccession,
+                           Set<Variant> unmatchedAccessions) {
+        stepContribution.setExitStatus(ExitStatus.COMPLETED);
+
+        if (!unmatchedAccessions.isEmpty()) {
+            stepContribution.setExitStatus(ExitStatus.FAILED);
+            logger.error("{} variants were found in the accession report that were not found in the original VCF.",
+                         unmatchedAccessions.size());
+            logger.info("These are the {} variants that were not found in the original VCF: {}",
+                        unmatchedAccessions.size(), unmatchedAccessions);
+        }
+
+        if (!variantsWithoutAccession.isEmpty()) {
+            stepContribution.setExitStatus(ExitStatus.FAILED);
+            logger.error("{} variants were not found in the accession report.", variantsWithoutAccession.size());
+            logger.info("These are the {} variants that were not found in the accession report: {}",
+                        variantsWithoutAccession.size(), variantsWithoutAccession);
+        }
+    }
+
+    public long getMaxVariantBufferSize() {
+        return maxVariantBufferSize;
+    }
+
+    /**
+     * See the terminology in the documentation of this class
+     * @param maxVariantBufferSize max size of the buffer where the variants of the original VCF are loaded
+     */
+    public void setMaxVariantBufferSize(long maxVariantBufferSize) {
+        this.maxVariantBufferSize = maxVariantBufferSize;
+    }
+
+    /**
+     * To grant correct behaviour in worst-ordering scenario, the accessionBuffer (for yet-unmatched accessions from
```

Comment:
* "guarantee" instead of "grant"
* "accession buffer" instead of "accessionBuffer" to allow the documentation to still stand if the variable name changes